### PR TITLE
Add an example for Client.Search() usage

### PR DESCRIPTION
**Pull request by pabluk**
I'm new in golang and some days ago I spent some hours trying to figure out how to use [Client.Search](https://godoc.org/github.com/emersion/go-imap/client#Client.Search) and finally I understand it but I also found that I wasn't the only one looking for that, see https://github.com/emersion/go-imap/issues/186#issuecomment-392602873 and https://github.com/emersion/go-imap/issues/172. So I decided to write a basic example of searching for unseen messages.

I try to follow the same style of other examples but please don't hesitate to tell if I need to change something and feel free to accept or reject this PR.

### CODE REVIEW

File: client/example_test.go [hunk 1]
```diff
@@ -218,3 +218,46 @@ func ExampleClient_Store() {
...
+	criteria := imap.NewSearchCriteria()
+	criteria.WithoutFlags = []string{imap.SeenFlag}
+	ids, err := c.Search(criteria)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Println("IDs found:", ids)
+
+	if len(ids) > 0 {
+
```
Comment (emersion) [inline client/example_test.go]: Style: extraneous line here

File: client/example_test.go [hunk 2]
```diff
@@ -218,3 +218,46 @@ func ExampleClient_Store() {
...
+		seqset := new(imap.SeqSet)
+		seqset.AddNum(ids...)
+
+		messages := make(chan *imap.Message, 10)
+		done := make(chan error, 1)
+		go func() {
+			done <- c.Fetch(seqset, []imap.FetchItem{imap.FetchEnvelope}, messages)
+		}()
+
+		log.Println("Last messages:")
```
[the remark this answers is not in the document]
Comment (pabluk) [reply client/example_test.go]: good catch! yes I'll change that.

File: client/example_test.go (hunk 1 above)
Comment (pabluk) [reply client/example_test.go]: oh yes I'll change that. Weird thing gofmt didn't complain about that.

**Comment by pabluk** [comment]
I've applied all the requested changes, I hope it's ok now.

Thanks for your time and feedback!

**Comment by emersion** [comment]
Thanks!

**Comment by zychappy** [comment]
it is a nice example which can println msg subject,but
when I try to get msg body,the result always nil
```go
var sect imap.BodySectionName
	for msg := range messages {
		fmt.Println("* " + msg.Envelope.Subject)
		r := msg.GetBody(&sect)
		if r == nil {
			panic("no body")
		}
...
```

**Pull request merged**